regression/regression_namelists.sh

```diff
@@ -18,7 +18,7 @@ export gsi_namelist="
    oneobtest=.false.,retrieval=.false.,l_foto=.false.,
    use_pbl=.false.,use_compress=.true.,nsig_ext=45,gpstop=50.,
    commgpstop=45.,commgpserrinf=1.0,
-   use_gfs_nemsio=.false.,lrun_subdirs=.true.,
+   use_gfs_nemsio=.false.,lrun_subdirs=.false.,
```

Comment:
Thanks for checking.  I wonder if external GSI partners use the system calls.   We still need to understand why `mkdir` from `gsi.x` is problematic with intel2022.

Reply:
Thank you @mark-a-potts for this suggestion.  Setting `lrun_subdir=.false.` is not desirable.  

With this option we write 6010 task, observation, and outer loop specific diagnostic files to the run directory of the hiproc = 144 task `global_4denvar` ctest.   Rerunning this test with `lrun_subdirs=.true.` creates 144 task specific sub-directories.   Each sub-directory contains a handful of diagnostic files associated with the given task.

The total file and directory count with `lrun_subdirs=.true.` in the hiproc `global_4denvar` ctest is 352.   When `lrun_subdirs=.false.` this increases to 6218.

If intel2022 requires us to abandon the `mkdir` option from `gsi.x`, workflow developers and GSI users will need to update scripts to either (a) create sub-directories at script level prior to executing `gsi.x` or (2) modify scripts and jobs which process GSI diagnostic files to switch from the task directory structure to a flat directory structure.  We want to avoid both of these options if at all possible.

Reply:
@mark-a-potts I will give that a shot.  Thanks!

@GeorgeVandenberghe-NOAA I can certainly open up another issue with Intel about this.  I will first try running with `lrun_subdir=.true.` to check out the errors and then see if Mark's suggestion works.  I'm curious if it could be an MPI/optimization bug.  If Mark's suggestion does work, I would suggest we open another issue while waiting for Intel support, since in my experience, it can take some time to hear back.

Reply:
@hu5970 I just ran 6 tests on Orion, with more in queue, with `lrun_subdir=.true.` and they all created directories without error using the `system` calls.  Perhaps this is a system issue or perhaps one of the other bug fixes resolve this?  Which system did you first notice these on and for which test(s)?

Reply:
Using the Intel portability function `MAKEDIRQQ` appears to have resolved this issue.  All of the `global_*var` tests ran successfully fully optimized and in debug mode.  I'm going to run the optimized suite once more today to make sure it's still working, since it is an intermittent failure, just to be sure.  I'll be submitting a PR to @hu5970's branch later today to include this fix assuming the tests go well.

Reply:
Unfortunately, I am still seeing some crashes when using `MAKEDIRQQ` on Hera.  However, I found another solution that works on all systems and does not require disabling Fortran-created directories by having just one process create all of the directories.

Reply:
I see `MAKEDIRQQ` in `obsmod.F90`.  I don't see the code limiting `MAKEDIRQQ` to a single task which creates directories for all tasks.   What am I missing?

Reply:
This is another case of hastiness.  I made this change and never committed it.  PR forthcoming.

Reply:
@RussTreadon-NOAA This is now fixed.  I opted to continue to use `MAKEDIRQQ` when using Intel compilers, but the directory creation is now handled on one task.